File: presentation/src/main/java/example/com/presentation/ui/main/MainFragment.java
package example.com.presentation.ui.main;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.Collection;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import example.com.presentation.R;
import example.com.presentation.models.GithubRepositoryModel;
import example.com.presentation.ui.base.BaseFragment;
import timber.log.Timber;

/**
 * Created by Mohammad Jafarzadeh Rezvan on 10/11/2018.
 */
public class MainFragment extends BaseFragment<MainPresenter, MainActivity>
        implements MainContract.View {

    @Inject ExampleAdapter mAdapter;

    private RecyclerView mRecyclerView;

    //---------------------------------------------------------------
    // Lifecycle
    //---------------------------------------------------------------

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getActivityComponent().inject(this);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        final View view = inflater.inflate(R.layout.fragment_main, container, false);
        mRecyclerView = view.findViewById(R.id.recycler_view);
        mRecyclerView.setAdapter(mAdapter);
        mRecyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mPresenter.loadData();
    }

    //---------------------------------------------------------------
    // MVP callbacks
    //---------------------------------------------------------------

    @Override
    public void githubRepositoriesLoaded(Collection<GithubRepositoryModel> githubRepositoryModels) {
        Timber.i("data loaded");
        mAdapter.setData(githubRepositoryModels);
    }

    @Override
    public void githubRepositoriesFailed(Throwable t) {

    }
}

File: presentation/src/main/java/example/com/presentation/ui/main/MainFragment.kt
package example.com.presentation.ui.main

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import example.com.presentation.R
import example.com.presentation.models.GithubRepositoryModel
import example.com.presentation.ui.base.BaseFragment
import timber.log.Timber
import javax.inject.Inject

/**
 * Created by Mohammad Jafarzadeh Rezvan on 10/11/2018.
 *
 */
class MainFragment : BaseFragment(), MainContract.View {

    @Inject internal lateinit var mAdapter: ExampleAdapter
    @Inject internal lateinit var mMainPresenter: MainPresenter

    private lateinit var mRecyclerView: RecyclerView

    private val mainActivity: MainActivity
        get() = this.activity as MainActivity? ?: throw RuntimeException("cannot cast activity")

    //---------------------------------------------------------------
    // Lifecycle
    //---------------------------------------------------------------

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mainActivity.activityComponent.inject(this)
    }

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View?
    {
        val view = inflater.inflate(R.layout.fragment_main, container, false)
        mRecyclerView = view.findViewById(R.id.recycler_view)
        mRecyclerView.adapter = mAdapter
        mRecyclerView.layoutManager = LinearLayoutManager(activity)
        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mMainPresenter.attach(this)
        mMainPresenter.loadData()
    }

    override fun onDestroyView() {
        super.onDestroyView()
        mMainPresenter.detach()
    }

    //---------------------------------------------------------------
    // MVP callbacks
    //---------------------------------------------------------------

    override fun githubRepositoriesLoaded(githubRepositoryModels: List<GithubRepositoryModel>) {
        Timber.i("data loaded")
        mAdapter.setData(githubRepositoryModels)
        mAdapter.notifyDataSetChanged()
    }

    override fun githubRepositoriesFailed(t: Throwable) {

    }
}
